raft/raft.go

```diff
@@ -730,7 +728,11 @@ func (r *raft) Step(m pb.Message) error {
 		default:
 			r.logger.Infof("%x [term: %d] received a %s message with higher term from %x [term: %d]",
 				r.id, r.Term, m.Type, m.From, m.Term)
-			r.becomeFollower(m.Term, lead)
```

Comment:
```go
if m.Type == msgApp || m.Type == msgSnap || m.Type == msgHeartbeat {
    r.becomeFollower(t.Term, m.From)
} else {
    r.becomeFollower(t.Term, None)
}
```

then we can remove all comments. i would like to make the code explicit to document itself instead of adding a bunch of comments.

Reply:
I think you meant:

if m.Type == pb.MsgApp || m.Type == pb.MsgSnap || m.Type == pb.MsgHeartbeat {
    r.becomeFollower(m.Term, m.From)
} else {
    r.becomeFollower(m.Term, None)
}

Reply:
yes.